### PR TITLE
Prepare VPG and others for MAML

### PR DESCRIPTION
**Pull request by naeioi**
This PR is part of the breakdown of the original MAML PR #1101 . It prepares existing codebase for MAML.

### CODE REVIEW

File: src/garage/torch/algos/vpg.py [hunk 1]
```diff
@@ -323,26 +332,37 @@ def process_samples(self, itr, paths):
...
+            pad_to_last(self._get_baselines(path),
+                        total_length=self.max_path_length) for path in paths
+        ])
 
-        return valids, obs, actions, rewards
+        return obs, actions, rewards, valids, baselines
 
-    def _log(self, itr, paths, loss_before, loss_after, kl_before, kl,
-             policy_entropy):
+    def evaluate_performance(self, itr, batch):
```
Comment (ryanjulian) [inline src/garage/torch/algos/vpg.py]: please call `RLAlgorithm.evaluate_performance` instead.

File: src/garage/torch/algos/vpg.py [hunk 2]
```diff
@@ -131,54 +131,56 @@ def train_once(self, itr, paths):
...
+                itr, {
+                    'paths': paths,
+                    'loss_before': loss.item(),
+                    'loss_after': loss_after.item(),
+                    'kl_before': kl_before.item(),
+                    'kl': kl.item(),
+                    'policy_entropy': policy_entropy.mean().item()
+                })
 
         self.baseline.fit(paths)
```
Comment (ahtsan) [inline src/garage/torch/algos/vpg.py]: Shall we fit baseline with the augmented reward? Are you using the original reward or augmented reward here?

Comment (ryanjulian) [reply src/garage/torch/algos/vpg.py]: I recommend you fit with augmented reward -- you want the baseline to be as accurate as possible. This only works if the baseline input has enough information to predict the augmented reward, though.

File: src/garage/torch/algos/vpg.py (hunk 1 above)
Comment (zequnyu) [reply src/garage/torch/algos/vpg.py]: Our intention for `evaluate_performance()` is not to override the function, but to call it with the batch.

Comment (naeioi) [reply src/garage/torch/algos/vpg.py]: How should I incorporate additional metrics, then? Metrics like `LossBefore, KL, Entropy` are not included in `np/algos/base.py: evaluate_performance()`.

File: src/garage/torch/algos/vpg.py (hunk 2 above)
Comment (naeioi) [reply src/garage/torch/algos/vpg.py]: Here, the original reward is fed to the baseline. This is inherited from the old implementation of torch/VPG. If we were to fix this I think we better make it in a separate PR.

File: src/garage/torch/algos/vpg.py (hunk 1 above)
Comment (ryanjulian) [reply src/garage/torch/algos/vpg.py]: those can be logged within the algorithm, but the standardized evaluation metrics should always be logged via `RLAlgorithm.evaluate_performance`. This makes sure they are all calculated the same across algorithms.

Comment (naeioi) [reply src/garage/torch/algos/vpg.py]: Are you suggesting that I override `evaluate_performance()` and call `RLAlgorithm.evaluate_performance()` in it, plus additional logs.

Comment (ahtsan) [reply src/garage/torch/algos/vpg.py]: maybe like calling `evaluate_performance()` like
https://github.com/rlworkgroup/garage/blob/master/src/garage/tf/algos/batch_polopt.py#L244
and log your own metrics like
https://github.com/rlworkgroup/garage/blob/master/src/garage/tf/algos/batch_polopt.py#L259

Comment (naeioi) [reply src/garage/torch/algos/vpg.py]: Thanks for point this out.

**Pull request merged**